search/everywhere_search_callback.cpp

```diff
@@ -0,0 +1,34 @@
+#include "search/everywhere_search_callback.hpp"
+
+#include "search/result.hpp"
+
+#include <cstddef>
+#include <utility>
+
+namespace search
+{
+EverywhereSearchCallback::EverywhereSearchCallback(Delegate & delegate, TOnResults onResults)
+  : m_delegate(delegate)
+  , m_onResults(std::move(onResults))
+{
+}
+
+void EverywhereSearchCallback::operator()(Results const & results)
+{
+  auto changedResults = results;
+
+  for (size_t i = 0; i < changedResults.GetCount(); ++i)
```

Comment:
No need to loop over and over again already emit results. Note, each time operator() is called, the same instance of results is passed to the operator() (it's guaranteed), only size of |results| increases.